### PR TITLE
use torch.matmul over torch.mm for coreml compatibility. Add apply_lo…

### PR DESCRIPTION
**Pull request by kells1986**
When I used `minLoRA` to train a model and convert it to coreml using coremltools I hit and error that `mm` was not recognized, switching to matmul fixes that error.

I also added a convenience function to loop through layers in a model and apply lora to layers whose name matches specified target names

### CODE REVIEW

File: minlora/model.py [hunk 1]
```diff
@@ -85,6 +85,13 @@ def apply_lora(layer, register=True, merge=False, lora_config=default_lora_confi
                 parametrize.remove_parametrizations(layer, attr_name, leave_parametrized=merge)
 
 
+def apply_lora_by_name(model, target_module_names, register=True, merge=False, lora_config=default_lora_config):
```
Comment (changjonathanc) [inline minlora/model.py]: This function is an extension to `add_lora`, so I think it's better to name it `add_lora_by_name` and put it under `add_lora`.

**Review by changjonathanc — CHANGES_REQUESTED** [review]
Thanks for the PR! Just need some minor changes and I'll merge it.

**Comment by kells1986** [comment]
changes made

**Review by changjonathanc — APPROVED** [review]
Thanks!

**Pull request merged**